serveradmin/powerdns/tests.py

```diff
@@ -0,0 +1,697 @@
+        """Test PowerDNS domain default type is NATIVE
+
+        When creating a new Serveradmin object of servertype domain the
+        default for type (if not specified) must be NATIVE.
+        """
+
+        domain_query = _create_object(DOMAIN_SERVERTYPE)
+        domain_name = domain_query.get()['hostname']
+        domain_type = domain_query.get()[DOMAIN_ATTR_TYPE]
+
+        self.assertEqual(domain_type, 'NATIVE',
+                         'Serveradmin default domain type is NATIVE.')
+        self.assertEqual(Domain.objects.get(name=domain_name).type, 'NATIVE',
+                         'PowerDNS default type is NATIVE')
+
+    def test_domain_created_with_type_master_is_master(self):
+        """Test Domain created with type MASTER is MASTER
+
+        When creating a new Serveradmin object of servertype domain with type
+        attribute set to MASTER the PowerDNS domain should also be of type
+        master.
+        """
+
+        domain_query = _create_object(DOMAIN_SERVERTYPE, type='MASTER')
+        domain_name = domain_query.get()['hostname']
+        domain_type = domain_query.get()[DOMAIN_ATTR_TYPE]
+
+        self.assertEqual(domain_type, 'MASTER',
+                         'Domain created with type MASTER is MASTER')
+        self.assertEqual(Domain.objects.get(name=domain_name).type, 'MASTER',
+                         'PowerDNS domain created with type MASTER is MASTER')
+
+    def test_domain_wrong_or_empty_type_fails(self):
+        """Test submitting a wrong type fails
+
+        When specifying a empty or wrong type for a domain this should already
+        fail while committing the new values.
+        """
+
+        domain_query = _create_object(DOMAIN_SERVERTYPE)
+
+        domain_query.update(type='')
+        with self.assertRaises(CommitError):
+            domain_query.commit(user=self.user)
+
+        domain_query.update(type='wrong')
+        with self.assertRaises(CommitError):
+            domain_query.commit(user=self.user)
+
+    def test_new_domain_creates_soa_record(self):
+        """Test domain SOA record creation
+
+        When a new domain is created it must create a SOA record.
+        """
+
+        domain_query = _create_object(DOMAIN_SERVERTYPE, soa=self.soa)
+        domain_id = domain_query.get()['object_id']
+
+        records = Record.objects.filter(domain=domain_id, type='SOA',
+                                        content=self.soa)
+        self.assertEqual(records.count(), 1,
+                         'PowerDNS domain creation also creates a SOA record')
+
+    def test_new_domain_creates_ns_record(self):
+        """Test domain NS record creation
+
+        When a new domain is created it must also create a NS record.
+        """
+
+        domain_query = _create_object(DOMAIN_SERVERTYPE, ns=self.ns)
+        domain_id = domain_query.get()['object_id']
+
+        records = Record.objects.filter(domain=domain_id, type='NS')
+        self.assertEqual(records.count(), 2,
```

Comment:
Might also be interesting to also assert before the action. So you definitely know that it was not 2 before, because of changes in the fixtures.

Reply:
I can't filter for the domain_id creation because I don't know it. And that somebody created a domain with the exact timestamp the tests run in future is pretty unlikely.

Reply:
Ah yeah, ok, makes sense. 👍